### ISSUE TITLE
Neural Embedding

### ISSUE THREAD

**Issue by luwalong** [gxyau/cat_feat_enc_challenge#3]
Goal: map each record to a space with 'smaller' dimension

We can separate the columns with *binary & nominal(1)* and *ordinal & day/month(2)*.

For the first one, we can make each column with one-hot encoding (having the dimension of the number of different values) and simply concatenate them. Also for the second one, we can digitize each ordinal and concatenate it.

I guess there should be different scales of feed-forward weight matrices, so suggest following architecture:
(1) -> one-hot(1) -> reduced(1) ->
(2) -> one-hot(2) -> reduced(2) -> combine(1)and(2) -> final logits -> binary classification
